src/github/client/github-client.ts

```diff
@@ -142,7 +142,7 @@ export class GitHubClient {
 				if (!config.headers) {
 					config.headers = {};
 				}
-				config.headers[apiKeyConfig.headerName] = await apiKeyConfig.apiKeyGenerator();
+				config.headers[apiKeyConfig.headerName] = await apiKeyConfig.apiKeyGenerator() ?? "undefined";
```

Comment:
We have already checked on line 139 that the apiKeyConfig is defined, which should mean that both headerName and the value are defined.